R/error-checks.R

```diff
@@ -25,21 +27,17 @@ isOfType <- function(object, type, nullAllowed = FALSE) {
 
```

Comment:
We can't use `validateIsLogical()` here because that function itself relies on `isOfType`, and so it leads to infinite recursion. 
I am therefore using a different approach.